inc/CrvHitInfoReco.hh

```diff
@@ -32,7 +33,10 @@ namespace mu2e
       nHits(nCoincidenceHits),
       nLayers(nCoincidenceLayers),
       angle(coincidenceAngle)
-    {}
+    {
+      // Copy PEsPerLayer array elements
+      for (int i = 0; i < 4; i++) this->PEsPerLayer[i] = PEsPerLayer[i];
```

Comment:
Is the number of CRV layers defined somewhere in the Offline? I'd prefer to use a named variable than a bare 4. Also, I prefer to have curly brackets for the for loop even if it's just one line for safety

Reply:
Good point. I had a quick look at there is a parameter `nLayers_` in `Offline/GeometryService/inc/CosmicRayShieldMaker.hh` which looks like a likely candidate. I will double check and add it in. I'll also add the curly brackets.

Reply:
You can use `CRVId::nLayers` defined in `Offline/DataProducts/inc/CRVId.hh`.

Reply:
Thanks Ralf

Reply:
Is it too late to define all of the c-style arrays of dimension 4  as std::array<int,4>?   Then you can do the copy in the intiailizer list:  PEsPerLayer(PEsPerLayer) .  One of the obscure but handy corners of c++ is that in an initializer list, the compiler knows that it means to have a data member and an argument with the same name

It it is too late definitely use std::array, then use CRVId::nLayers from Offline/DataProducts/inc/CRVId.hh  .  That's a compile time condstant  rather than a variable that you have to look up at run time.

I don't think that the recommendation of Offline/GeometryService/inc/CosmicRayShieldMaker.hh will work.  Do not try to make an instance of that class yourself.  It is to only be using insde the geometry service.

Reply:
Hi Rob. Thanks for the comment. It's not too late to define the array as a `std::array<int, CRVId::nLayers>` and then copy it to the initializer as `PEsPerLayer(PEsPerLayer)`, I will add that in. Ralf's recommendation to use CRVId::nLayers from `Offline/DataProducts/inc/CRVId.hh` works well, so I won't attempt to use anything from `CosmicRayShieldMaker.hh`.

Reply:
Although I suppose it should be an array of float rather than ints.

Reply:
It seems like you need to copy the array elements using curly brackets: `PEsPerLayer{PEsPerLayer}` rather than `PEsPerLayer(PEsPerLayer)`.